packages/@storefront/flux-capacitor/test/unit/flux-capacitor.ts

```diff
@@ -99,29 +99,178 @@ suite('FluxCapacitor', ({ expect, spy, stub }) => {
+
+          flux.saveState(route);
+
+          expect(pushState).to.be.calledWith({ route });
+        });
+      });
+
+      describe('pushState()', () => {
+        it('should call updateHistory', () => {
+          const route = 'search';
+          const url = 'www.url.com';
+          const updateHistory = flux.updateHistory = spy();
+
+          flux.pushState({ route, url });
+
+          expect(updateHistory).to.be.calledWith({ shouldFetch: true, url, route, method: flux.history.pushState });
+        });
+
+        it('should allow shouldFetch to be overriden', () => {
+          const route = 'search';
+          const url = 'www.url.com';
+          const shouldFetch = false;
+          const updateHistory = flux.updateHistory = spy();
+
+          flux.pushState({ shouldFetch, route, url });
+
+          expect(updateHistory).to.be.calledWith({ shouldFetch, url, route, method: flux.history.pushState });
+        });
+      });
+
+      describe('replaceState()', () => {
+        it('should dispatch updateHistory directly when buildAndParse is not passed in', () => {
+          const route = 'details';
+          const FETCH = 'fetch';
+          const updateHistory = stub()
+            .withArgs({ shouldFetch: false, route, method: flux.history.replaceState })
+            .returns(FETCH);
+          const dispatch = spy();
+          flux.actions = <any>{ updateHistory };
+          flux.store = <any>{ dispatch };
+
+          flux.replaceState(route);
+
+          expect(dispatch).to.be.calledWith(FETCH);
+        });
+
+        it('should dispatch updateHistory directly when buildAndParse is false', () => {
+          const route = 'details';
+          const FETCH = 'fetch';
+          const updateHistory = stub()
+            .withArgs({ shouldFetch: false, route, method: flux.history.replaceState })
+            .returns(FETCH);
+          const dispatch = spy();
+          flux.actions = <any>{ updateHistory };
+          flux.store = <any>{ dispatch };
 
-        flux.replaceState(route);
+          flux.replaceState(route, false);
 
-        expect(emit).to.be.calledWith(Events.HISTORY_REPLACE, { state, route });
+          expect(dispatch).to.be.calledWith(FETCH);
+        });
+
+        it('should call updateHistory when buildAndParse is true', () => {
+          const route = 'details';
+          const updateHistory = flux.updateHistory = spy();
+
+          flux.replaceState(route, true);
+
+          expect(updateHistory).to.be.calledWith({ shouldFetch: false, route, method: flux.history.replaceState });
+        });
+      });
+
+      describe('updateHistory()', () => {
+        it('should parse the url and dispatch updateHistory', (done) => {
+          const url = 'www.url.com';
+          const shouldFetch = true;
+          const method = () => null;
+          const FETCH = 'fetch';
+          const route = 'search';
+          const request = { a: 'b' };
+          const dispatch = spy(() => {
```

Comment:
This feels odd, but I'm not sure there's a better way to do it unless you return the promise from `flux.updateHistory`.

Reply:
Any thoughts on how to make this a better test?

Reply:
Nope.  An awkward test usually means the thing being tested could be written better, but I'm not sure it's worth it.